src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -82,6 +84,13 @@ static <T> Blob constructBlobAndMetadata(
       ChannelFlushContext firstChannelFlushContext =
           channelsDataPerTable.get(0).getChannelContext();
 
+      EncryptionKey encryptionKey =
```

Comment:
For now, we would fall back to the channel flush context. But after we remove our old logic, i guess we would have to throw some kind of error?

Reply:
I mean we should always find the key, right? If we don't, it is an error of some kind, which we should report as an internal error. The code I see here would lead to an NPE.